askama_derive/src/tests.rs

```diff
@@ -1410,3 +1414,117 @@ fn test_bare_cr_doc_comment() -> Result<(), syn::Error> {
 
     Ok(())
 }
+
+#[test]
+fn test_conditional_expr() {
+    compare(
+        r"{{ a if b else c }}",
+        r#"
+        match (
+            &((&&askama::filters::AutoEscaper::new(
+                &(if askama::helpers::as_bool(&(self.b)) { self.a } else { self.c }),
+                askama::filters::Text,
+            ))
+                .askama_auto_escape()?),
+        ) {
+            (expr0,) => {
+                (&&&askama::filters::Writable(expr0))
+                    .askama_write(__askama_writer, __askama_values)?;
+            }
+        }"#,
+        &[("a", "i8"), ("b", "i8"), ("c", "i8")],
+        3,
+    );
+
+    compare(
+        r"{{ a if b is defined else c }}",
+        r#"
+        match (
+            &((&&askama::filters::AutoEscaper::new(&(self.a), askama::filters::Text))
+                .askama_auto_escape()?),
+        ) {
+            (expr0,) => {
+                (&&&askama::filters::Writable(expr0))
+                    .askama_write(__askama_writer, __askama_values)?;
+            }
+        }"#,
+        &[("a", "i8"), ("b", "i8"), ("c", "i8")],
+        3,
+    );
+
+    compare(
+        r"{{ a if b is not defined else c }}",
+        r#"
+        match (
+            &((&&askama::filters::AutoEscaper::new(&(self.c), askama::filters::Text))
+                .askama_auto_escape()?),
+        ) {
+            (expr0,) => {
+                (&&&askama::filters::Writable(expr0))
+                    .askama_write(__askama_writer, __askama_values)?;
+            }
+        }"#,
+        &[("a", "i8"), ("b", "i8"), ("c", "i8")],
+        3,
+    );
+
+    compare(
+        r"{{ a if b if c else d else e }}",
+        r#"
+        match (
+            &((&&askama::filters::AutoEscaper::new(
+                &(if askama::helpers::as_bool(
+                    &(if askama::helpers::as_bool(&(self.c)) { self.b } else { self.d })
+                ) {
+                    self.a
+                } else {
+                    self.e
+                }),
+                askama::filters::Text,
+            ))
+                .askama_auto_escape()?),
+        ) {
+            (expr0,) => {
+                (&&&askama::filters::Writable(expr0))
+                    .askama_write(__askama_writer, __askama_values)?;
+            }
+        }"#,
+        &[
+            ("a", "i8"),
+            ("b", "i8"),
+            ("c", "i8"),
+            ("d", "i8"),
+            ("e", "i8"),
+        ],
+        3,
+    );
+
+    compare(
+        r"{{ a if b else c if d else e }}",
```

Comment:
This is absolutely horrifying my brain to parse. :rofl:

Reply:
What about this formatting? :)
```jinja
{{
   a if b else
   c if d else
   e
}}
```

Reply:
Yes, that does work.